agcaballero/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -73,15 +76,29 @@ public String toString() {
     /**
      * @return a Comment as a HTML div with proper formatting to be displayed
      */
-    private String htmlFormat() {
+    private String htmlFormat(String languageCode) {
       // gets time in local time zone (default: US ET)
       LocalDateTime localTime =
           new LocalDateTime(timestamp.getTime(), DateTimeZone.forID("US/Eastern"));
       DateTimeFormatter formatter = DateTimeFormat.forPattern("h:mm a M/dd/yy");
 
       return "<div class='comment-div'>"
           + "<p class='date'>" + formatter.print(localTime) + "</p>"
-          + "<p><b>" + name + " (" + email + "):</b> <br><br>" + text + "</p></div>";
+          + "<p><b>" + name + " (" + email + "):</b></p>"
+          + "<p class='comment-text' lang=" + languageCode + ">" + text + "</p></div>";
+    }
+
+    /**
+     * Translate the text to the language of a given language code
+     */
+    public String translateText(String languageCode) {
+      // Do the translation.
+      Translate translate = TranslateOptions.getDefaultInstance().getService();
+      Translation translation =
+          translate.translate(text, Translate.TranslateOption.targetLanguage(languageCode));
```

Comment:
Is the translate service able to figure out what the current language is? For example, if text is "red", then it's translated to French "rouge", and then translated to some other language, how does the Translate service know the current text is French and not English (it would mean the different between "red" and a type of makeup).

Reply:
I have tried this experimentally (e.g. entering text in Spanish and English and translating both to another language like French). It will translate the text properly, so it can figure out the current language. I think it just detects the current language.

Reply:
Maybe this wasn't a good example, and one of these would be a better test:
http://transpremium.com/same-word-different-meaning-englishfrench/

For example, if the word is "chair", which means different things in English and French, and you translate to a different language, which meaning would you end up with?

 it seems like one option for maintaining accuracy would be to also set a sourceLanguage in the TranslationOption:
https://googleapis.dev/java/google-cloud-clients/0.110.0-alpha/com/google/cloud/translate/Translate.TranslateOption.html

Reply:
I agree. I think the best course of action would just be to let the user input the language via dropdown they're typing in, and storing that in the comment as the source language.

Reply:
I tried it with chair, and it works now because I began storing the source language.

Reply:
Done